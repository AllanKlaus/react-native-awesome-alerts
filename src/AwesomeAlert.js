import React, {Component} from 'react';

import Alert from './containers';

export default class AwesomeAlert extends Component {

  _renderAlert = () => {
    return (
      <Alert {...this.props} />
    );
  };

  render() {
    return this._renderAlert();
  };
};
