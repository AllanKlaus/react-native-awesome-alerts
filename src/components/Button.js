import React, {Component} from 'react';
import {View, TouchableOpacity, Text} from 'react-native';

import PropTypes from 'prop-types';

import styles from './styles/ButtonStyles';

export default class Button extends Component {

  _renderButton = () => {
    const {text, backgroundColor} = this.props;
    const {onPress} = this.props;

    const containerStyle = {backgroundColor};

    return (
      <TouchableOpacity
        onPress={onPress}
          >
        <View style={[styles.container, containerStyle]}>
          <Text style={styles.text}>{text}</Text>
        </View>
    </TouchableOpacity>
    );
  };

  render() {
    const {show} = this.props;

    if(show)
      return this._renderButton();
    else
      return null
  };
};

Button.propTypes = {
  show: PropTypes.bool,
  text: PropTypes.string,
  backgroundColor: PropTypes.string
};

Button.defaultProps = {
  show: false
};
