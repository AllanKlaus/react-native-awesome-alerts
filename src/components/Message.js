import React, {Component} from 'react';
import {View, Text} from 'react-native';

import PropTypes from 'prop-types';

import styles from './styles/MessageStyles';

export default class Message extends Component {

  _renderMessage = (message) => {
    return (
      <View style={styles.container}>
        <Text style={styles.text}>{message}</Text>
      </View>
    );
  }

  render() {
    const {message} = this.props;
    const show = typeof message === 'string';

    if(show)
      return this._renderMessage(message);
    else
      return null;
  };
};

Message.propTypes = {
  message: PropTypes.string
};
