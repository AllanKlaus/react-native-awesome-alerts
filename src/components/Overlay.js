import React, {Component} from 'react';
import {View, TouchableWithoutFeedback} from 'react-native';

import PropTypes from 'prop-types';

import styles from './styles/OverlayStyles';

export default class Overlay extends Component {

  render() {
    const {onPress} = this.props;

    return (
      <TouchableWithoutFeedback onPress={onPress} >
        <View style={styles.container} />
      </TouchableWithoutFeedback>
    );
  };
};

Overlay.propTypes = {
  onPress: PropTypes.func
};
