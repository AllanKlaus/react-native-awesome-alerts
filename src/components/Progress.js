import React, {Component} from 'react';
import {ActivityIndicator} from 'react-native';

import PropTypes from 'prop-types';

export default class Progress extends Component {

  _renderProgress = () => {
    return (
      <ActivityIndicator />
    );
  };

  render() {
    const {showProgress} = this.props;

    if(showProgress)
      return this._renderProgress();
    else
      return null;
  };
};

Progress.propTypes = {
  showProgress: PropTypes.bool
};
