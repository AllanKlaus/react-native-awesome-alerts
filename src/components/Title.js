import React, {Component} from 'react';
import {View, Text} from 'react-native';

import styles from './styles/TitleStyles';

export default class Title extends Component {

  _renderTitle = (title) => {
    return (
      <View style={styles.container}>
        <Text style={styles.text}>{title}</Text>
      </View>
    );
  };

  render() {
    const {title} = this.props;
    const show = typeof title === 'string';

    if(show)
      return this._renderTitle(title);
    else
      return null;
  };
};
