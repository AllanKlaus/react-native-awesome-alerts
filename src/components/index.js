import Progress from './Progress';
import Title from './Title';
import Message from './Message';
import Button from './Button';
import Overlay from './Overlay';

export {
  Progress,
  Title,
  Message,
  Button,
  Overlay
};
