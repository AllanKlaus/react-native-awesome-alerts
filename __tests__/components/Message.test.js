import 'react-native';
import React from 'react';

import { Message } from '../../src/components';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(
    <Message message="I am just a message" />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
