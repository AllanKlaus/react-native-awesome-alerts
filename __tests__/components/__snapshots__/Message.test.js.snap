// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "paddingTop": 5,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#7b7b7b",
        "fontSize": 14,
      }
    }
  >
    I am just a message
  </Text>
</View>
`;
